daemon/SecretsImpl/pluginfunctionwrappers_p.h

```diff
@@ -0,0 +1,296 @@
+#include "Secrets/extensionplugins.h"
+#include "Secrets/secretmanager.h"
+#include "Secrets/result.h"
+
+#include <QtCore/QString>
+#include <QtCore/QByteArray>
+
+#include <tuple>
+
+namespace Sailfish {
+
+namespace Secrets {
+
+namespace Daemon {
+
+namespace ApiImpl {
+
+std::tuple<bool, // found
+           Sailfish::Secrets::Result> lockSpecificPlugin(
+        const QMap<QString, Sailfish::Secrets::StoragePlugin*> &storagePlugins,
+        const QMap<QString, Sailfish::Secrets::EncryptionPlugin*> &encryptionPlugins,
+        const QMap<QString, Sailfish::Secrets::EncryptedStoragePlugin*> &encryptedStoragePlugins,
+        const QString &lockCodeTarget);
+
+std::tuple<bool, // found
+           Sailfish::Secrets::Result> unlockSpecificPlugin(
+        const QMap<QString, Sailfish::Secrets::StoragePlugin*> &storagePlugins,
+        const QMap<QString, Sailfish::Secrets::EncryptionPlugin*> &encryptionPlugins,
+        const QMap<QString, Sailfish::Secrets::EncryptedStoragePlugin*> &encryptedStoragePlugins,
+        const QString &lockCodeTarget,
+        const QByteArray &lockCode);
+
+std::tuple<bool, // found
+           Sailfish::Secrets::Result> modifyLockSpecificPlugin(
+        const QMap<QString, Sailfish::Secrets::StoragePlugin*> &storagePlugins,
+        const QMap<QString, Sailfish::Secrets::EncryptionPlugin*> &encryptionPlugins,
+        const QMap<QString, Sailfish::Secrets::EncryptedStoragePlugin*> &encryptedStoragePlugins,
+        const QString &lockCodeTarget,
+        std::tuple<QByteArray, QByteArray> newAndOldLockCode);
+
+bool lockPlugins(
+        QList<Sailfish::Secrets::EncryptionPlugin*> encryptionPlugins,
+        QList<Sailfish::Secrets::StoragePlugin*> storagePlugins,
+        QList<Sailfish::Secrets::EncryptedStoragePlugin*> encryptedStoragePlugins);
+
+bool unlockPlugins(
+        QList<Sailfish::Secrets::EncryptionPlugin*> encryptionPlugins,
+        QList<Sailfish::Secrets::StoragePlugin*> storagePlugins,
+        QList<Sailfish::Secrets::EncryptedStoragePlugin*> encryptedStoragePlugins,
+        const QByteArray &encryptionKey);
+
+bool modifyLockPlugins(
+        QList<Sailfish::Secrets::EncryptionPlugin*> encryptionPlugins,
+        QList<Sailfish::Secrets::StoragePlugin*> storagePlugins,
+        QList<Sailfish::Secrets::EncryptedStoragePlugin*> encryptedStoragePlugins,
+        const QByteArray &oldEncryptionKey,
+        const QByteArray &newEncryptionKey);
+
+namespace EncryptionPluginWrapper {
+    bool isLocked(Sailfish::Secrets::EncryptionPlugin *plugin);
+    bool lock(Sailfish::Secrets::EncryptionPlugin *plugin);
+    bool unlock(Sailfish::Secrets::EncryptionPlugin *plugin,
+                const QByteArray &lockCode);
+    bool setLockCode(Sailfish::Secrets::EncryptionPlugin *plugin,
+                     const QByteArray &oldLockCode,
+                     const QByteArray &newLockCode);
+    std::tuple<Sailfish::Secrets::Result, QByteArray> deriveKeyFromCode(
```

Comment:
This contradicts the current sailfish-secrets style, where we pass additional output arguments as pointers. I would suggest to change the signature to return a `Result` (like everything else), and add the `QByteArray` as a pointer argument.

Reply:
QtConcurrent::run() has a limit of 5 arguments to a given method, and pointer outparams are a BAD idea for concurrent code.  Hence the tuplisation and outparams done this way.